### DISCUSSION TITLE
Planet-making tutorials from basics to complex concepts

### DISCUSSION THREAD

**Discussion by pkmniako** [Kopernicus/wiki#73] [General]
It would be a good fit for the wiki to have a series of tutorials that would guide a completely new user in how to make a planet with Kopernicus, from the very basics to something that resembles a playable world, to be planned and written by the community.

These tutorials would **build upon previous tutorials** so progress is not broken up into varied and separated sections which users would need to navigate.

This could be done in separate sections:

- **Pre-requisites**: Small optional tutorials to help absolutely new users on how to work with `cfg` files, what the `GameData` folder is and similar concepts, just so everyone can start the actual planet-making on the same level
- **Beginners**: A series of **linear** tutorials, one building upon the previous one, that have the user go from spawning a clone of a stock body to having a new, airless body with some wiki-provided textures (Mars???).
- **Intermediate**: A series of more branching tutorials helping with additional but popular concepts, like Oceans, Atmospheres, Vertex Height Mods, Rings, etc.
- **Advanced**: A series of tutorials on more niche and/or complex topics, like for example binary planets, stars, system replacers, and more.

For this series of tutorials to be done by the community, it is **extremely important** to plan out the different sections/tutorials and how to link them with previous ones, before writing any sentence. Input on what these sections/tutorials should be, specially for the **Beginners** section, is appreciated.

**Comment by pkmniako** [comment]
As a first opinion, I think a nice, non-complete selection of tutorials for the **Beginners** section could be:

- Create a clone of an existing planet, but at a different orbit (Templates?)
- Modifying the properties of your planet (Change size, mass, orbit, etc. Basic stuff)
- Adding a heightmap (Teaching how heightmaps work, how to add them to a planet.)
  - Give user an already-done texture, like Mars' Heightmap, so they don't have to deal with making their own
  - Revolve the tutorial around having the user teleport to specific coordinates so they can compare the results with the ones shown in the wiki
- Adding a colormap (Teaching how colormaps work, how to add them to a planet. Give user an already-done texture, like Mars')
  - Same as the previous section, give them an already-done texture, like Mars' Colormap
- Updating scaledspace (How to make the planet look like you want from space)
- Surface Material
- Normal maps, how to generate them and use them

**Reply by DeltaDizzy** [reply]
Yeah I think this would be really good! Though, you have to do a bit of property changes in the first one to change the orbit...

**Reply by pkmniako** [reply]
> Yeah I think this would be really good! Though, you have to do a bit of property changes in the first one to change the orbit...

Agree, though in the first point, the tutorial could force the values and ask the user to trust the tutorial, with the second point then explaining what it does

**Reply by pkmniako** [reply]
I added "Surface Material" to the list, as it dictates very heavily how the surface looks, which I believe is pretty important for a basic planet

**Reply by DeltaDizzy** [reply]
> which I believe is pretty important for a basic planet

I feel like that depends on our definition of "basic". Do we mean "a functioning planet" or "a planet that could be a part of a mod that nobody would raise an eyebrow at, but just does not use many features"? I am tempted to favor the former just because the definition of what specific extra features fall within basic is very arbitrary. What about larger PQS mods? Oceans? Atmospheres? Surface material can be an early intermediate topic but I feel like it is a little beyond scope for an introductory "how to make a planet" series.

**Comment by cloverdove** [comment]
where would a hugin tutorial slot in? just a fast video explaining it would be very very helpful somewhere in the beginners section

**Reply by DeltaDizzy** [reply]
We should probably define where "tools and techniques" tutorials go, be that its own category, advanced, intermediate, or distributed between all categories (I really dont think putting _all_ tool tutorials in prereqs or beginners is best given how complicated some can be and how optional they are). I personally favor a separate category linked to from the regular set of tutorials where needed but it is very much up for debate.
